### PR TITLE
[libtheora] install pkg-config files

### PR DESCRIPTION
**Pull request by luncliff**
### Changes

Generate/Install `.pc` files for pkg-config users.

### References

* https://github.com/microsoft/vcpkg/pull/28204 (previous update)

### Checklist

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libtheora/CMakeLists.txt [hunk 1]
```diff
@@ -141,3 +142,17 @@ install(EXPORT unofficial-theora-targets
   NAMESPACE unofficial::theora::
   DESTINATION "lib/unofficial-theora"
 )
+
+set(prefix "${CMAKE_INSTALL_PREFIX}")
+set(libdir "\${prefix}/lib")
+set(exec_prefix "\${prefix}")
+set(includedir "\${prefix}/include")
+set(VERSION "1.2.0-alpha1")
```
Comment (dg0yt) [inline ports/libtheora/CMakeLists.txt:150]: Maintenance trap. Can this follow some variable? If nothing is available in the project, derive from `VERSION` in portfile.

Comment (luncliff) [reply ports/libtheora/CMakeLists.txt:150]: I will check if other ports have similar patterns or solutions.

File: ports/libtheora/CMakeLists.txt [hunk 2]
```diff
@@ -141,3 +142,16 @@ install(EXPORT unofficial-theora-targets
...
+set(prefix "${CMAKE_INSTALL_PREFIX}")
+set(libdir "\${prefix}/lib")
+set(exec_prefix "\${prefix}")
+set(includedir "\${prefix}/include")
+configure_file(theora.pc.in theora.pc @ONLY)
+configure_file(theoradec.pc.in theoradec.pc @ONLY)
+configure_file(theoraenc.pc.in theoraenc.pc @ONLY)
+
+install(
+  FILES ${PROJECT_BINARY_DIR}/theora.pc ${PROJECT_BINARY_DIR}/theoradec.pc ${PROJECT_BINARY_DIR}/theoraenc.pc
```
Comment (dg0yt) [inline ports/libtheora/CMakeLists.txt:155]:
```suggestion
  FILES "${PROJECT_BINARY_DIR}/theora.pc" "${PROJECT_BINARY_DIR}/theoradec.pc" "${PROJECT_BINARY_DIR}/theoraenc.pc"
```

File: ports/libtheora/portfile.cmake [hunk 3]
```diff
@@ -21,12 +21,13 @@ endif()
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
+        -DVERSION:STRING="${VERSION}"
```
Comment (dg0yt) [inline ports/libtheora/portfile.cmake:24]:
```suggestion
        "-DVERSION:STRING=${VERSION}"
```
Quoting for CMake, not for the shell.

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/libtheora/CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.0)
 project(theora LANGUAGES C)
+include(GNUInstallDirs)
```
Comment (dg0yt) [inline ports/libtheora/CMakeLists.txt:3]:
```suggestion
```
Probably unused.

**Review by FrankXie05 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**